fjord/suggest/providers/trigger/provider.py

```diff
@@ -0,0 +1,112 @@
+import logging
+
+from fjord.base.google_utils import ga_track_event
+from fjord.redirector import (
+    Redirector,
+    RedirectParseError,
+    build_redirect_url
+)
+from fjord.suggest import Link, Suggester
+from fjord.suggest.providers.trigger.models import TriggerRule
+
+
+PROVIDER = 'trigger'
+PROVIDER_VERSION = 1
+
+TRIGGER = 'tgr'
+EVENT_CATEGORY = 'trigger' + str(PROVIDER_VERSION)
+SESSION_KEY = 'trigger_{0}'
+
+
+logger = logging.getLogger('i.trigger')
+
+
+def format_redirect(slug):
+    return '{0}.{1}'.format(TRIGGER, slug)
+
+
+def parse_redirect(redirect):
+    try:
+        trigger, slug = redirect.split('.')
+        return trigger, slug
+    except IndexError as exc:
+        raise RedirectParseError(str(exc))
+
+
+def build_ga_category(rule):
+    return 'trigger_{rule.slug}_{rule.id}'.format(rule=rule)
+
+
+class TriggerRedirector(Redirector):
+    """Provides redirection urls"""
+    def handle_redirect(self, request, redirect):
+        # If it's not a trigger redirect, return.
+        if not redirect.startswith(TRIGGER):
+            return
+
+        # Get the response id for the last feedback left.
+        response_id = request.session.get('response_id', None)
+        if response_id is None:
```

Comment:
Do all the redirectors do this? Is it too late to make the response a parameter that this function takes, instead of having to pull it out of the session?